design-documents/graph-ql/coverage/Cart.graphqls

```diff
@@ -11,22 +11,134 @@ type CartQueryOutput {
+    email: String
+    shipping_addresses: [ShippingCartAddress]!
+    billing_address: BillingCartAddress
+    available_payment_methods: [AvailablePaymentMethod] @doc(description: "Available payment methods")
+    selected_payment_method: SelectedPaymentMethod
+    prices: CartPrices
+    total_quantity: Float!
+    is_virtual: Boolean!
+}
+
+type AvailablePaymentMethod {
+    code: String! @doc(description: "The payment method code")
+    title: String! @doc(description: "The payment method title.")
+}
+
+type SelectedPaymentMethod {
+    code: String! @doc(description: "The payment method code")
+    title: String! @doc(description: "The payment method title.")
+    purchase_order_number: String @doc(description: "The purchase order number.")
+}
+
+interface CartItemInterface @typeResolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CartItemTypeResolver") {
     id: String!
+    quantity: Float!
+    prices: CartItemPrices
+    product: ProductInterface!
+    customizable_options: [SelectedCustomizableOption]
+}
+
+type SimpleCartItem implements CartItemInterface @doc(description: "Simple Cart Item") {
+}
 
-    line_items_count: Int!
-    items_quantity: Float!
+type VirtualCartItem implements CartItemInterface @doc(description: "Virtual Cart Item") {
+}
 
-    selected_payment_method: CheckoutPaymentMethod
-    available_payment_methods: [CheckoutPaymentMethod]!
+type ConfigurableCartItem implements CartItemInterface {
+    child_sku: String! @doc(description: "SKU of the simple product corresponding to a set of selected configurable options.")
+    configurable_options: [SelectedConfigurableOption!]
+}
 
-    customer: CheckoutCustomer
-    customer_notes: String
+type DownloadableCartItem implements CartItemInterface @doc(description: "Downloadable Cart Item") {
+    links: [DownloadableProductLinks] @deprecated(description: "Type was renamed from plural to singular, also link now has different ID type")
+    links_v2: [DownloadableProductLink] @doc(description: "An array containing information about the selected links")
+    samples: [DownloadableProductSamples] @doc(description: "DownloadableProductSamples defines characteristics of a downloadable product")
+}
 
-    gift_cards_amount_used: Money
-    applied_gift_cards: [CartGiftCard]
+type BundleCartItem implements CartItemInterface {
+    bundle_options: [SelectedBundleOption!]
+}
 
-    is_multishipping: Boolean!
-    is_virtual: Boolean!
+type GiftCardCartItem implements CartItemInterface {
+    sender_name: String!
+    recepient_name: String!
+    amount: SelectedGiftCardAmount
+    message: String
+}
+
+type SelectedConfigurableOption {
+    id_v2: ID! @doc(description: "Hash which includes option ID, option type and slected values. Can be used to move configurable product to wishlist or gift registry")
```

Comment:
Can you explain the reason for both `id` and `id_v2` here? Not super clear to me

Reply:
`id` was there before and is actual integer value of the option ID.

`id_v2` is a hash which contains enough information to specify this option selection when adding product to cart/wishlist/gift registry. Data which may be contained here (depending on the option):
- Option type: custom/bundle/configurable/gift card amount etc
- Option ID
- Selected option value
- Maybe, entered value for the gift card (this is not finalized yet)

Reply:
Why does the client need to know what the ID consists of? It sounds like something that needs to be known by whoever is implementing the schema on the server

Reply:
The client does not, but implementors do. It is much easier to describe it here for whoever will be working on this schema. We can skip this description in the schema committed to the Magento repo.

Reply:
Sure, I'm just curious why we wouldn't use an inline comment? That's what comments are part of SDL for :)

Reply:
Would also still like to see a name that better reflects what `id_v2` means for a client - you've specified intent for the field, but that doesn't show through in the schema design yet

Reply:
(nevermind, we're discussing this in another thread - too many schema reviews!)